internal/server/smb/server.go

```diff
@@ -5,36 +5,63 @@ import (
 	"fmt"
 
 	"github.com/kubernetes-csi/csi-proxy/client/apiversion"
+	fsserver "github.com/kubernetes-csi/csi-proxy/internal/server/filesystem"
 	"github.com/kubernetes-csi/csi-proxy/internal/server/smb/internal"
+	"k8s.io/klog"
 )
 
 type Server struct {
-	hostAPI API
+	hostAPI  API
+	fsServer *fsserver.Server
 }
 
 type API interface {
-	NewSmbGlobalMapping(remotePath, username, password string) error
+	IsSmbMapped(remotePath string) (bool, error)
+	SMBLink(remotePath, localPath string) error
```

Comment:
maybe a API name more with more details? Like CreateRemoteShareLink?

Reply:
To make is similar to the other APIs - NewSmbLlink